mne/io/eeglab/eeglab.py

```diff
@@ -253,15 +254,40 @@ def _set_dig_montage_in_init(self, montage):
         self.set_montage(montage + make_dig_montage(ch_pos=ch_pos, coord_frame="head"))
 
 
-def _handle_montage_units(montage_units):
-    n_char_unit = len(montage_units)
-    if montage_units[-1:] != "m" or n_char_unit > 2:
-        raise ValueError(
-            '``montage_units`` has to be in prefix + "m" format'
-            f', got "{montage_units}"'
-        )
+def _handle_montage_units(montage_units, eeg):
+    if montage_units == "auto":
+        # estimate units from the channel position value range
+        # units could be mm, cm, or m
+        if (
+            hasattr(eeg, "chanlocs")
+            and "X" in eeg.chanlocs
+            and "Y" in eeg.chanlocs
+            and "Z" in eeg.chanlocs
+            and len(np.ravel(eeg.chanlocs["X"])) > 0
+        ):
+            xyz = np.array(
+                [
+                    np.ravel(a)
+                    for a in [eeg.chanlocs["X"], eeg.chanlocs["Y"], eeg.chanlocs["Z"]]
+                ]
+            ).T
+            is_nan_locs = np.isnan(xyz).any(axis=1)
+            mean_radius = np.mean(np.linalg.norm(xyz[~is_nan_locs], axis=1))
+            # radius should be between 0.05 and 0.11 meters
+            if mean_radius < 0.25:  # m
+                prefix = ""
+            elif mean_radius < 2.5:  # dm
+                prefix = "d"
+            elif mean_radius > 25:  # mm
+                prefix = "m"
+            else:  # 2.5 <= mean_radius <= 25, cm
+                prefix = "c"
+        else:
+            prefix = "m"  # assume mm if no channel positions are available
+    else:
+        _check_option("montage_units", montage_units, ("m", "dm", "cm", "mm"))
 
-    prefix = montage_units[:-1]
+        prefix = montage_units[:-1]
```

Comment:
And this can be restored
```suggestion
    prefix = montage_units[:-1]
```